### DISCUSSION TITLE
Cannot override /docs when replacing favicon

### DISCUSSION THREAD

**Discussion by fritz-trawa** [fastapi/fastapi#12083] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.openapi.docs import get_swagger_ui_html

# this should be enough to replace the favicon - but it isn't
app = FastAPI(
  title="blah",
  swagger_ui_parameters={"swagger_favicon_url": "/favicon.png"},
  # openapi_url="" # if I add this, then /docs will show the favicon I want, but none of the content: 'Failed to load API definition'
)

# tested accessing it on its own and it works
@app.get("/favicon.png", include_in_schema=False)
async def favicon():
    return FileResponse("./a/path.png")

# tested and it works - I get the openapi docs with the favicon
@app.get("/docs2", include_in_schema=False)
def overridden_swagger():
    return get_swagger_ui_html(
        openapi_url="/openapi.json",
        title="DOES IT EVEN WORKS",
        swagger_favicon_url="/favicon.png",
    )

# doesn't work - I get the standard docs
# tried both async def and just def
@app.get("/docs", include_in_schema=False)
def overridden_swagger():
    return get_swagger_ui_html(
        openapi_url="/openapi.json",
        title="OVERRIDDEN!",
        swagger_favicon_url="/favicon.png",
    )
```

### Description

I am trying to replace the favicon for the openapi docs, but it doesn't work.

`swagger_ui_parameters` is ignored when instantiating the app, and when I create a `/docs` route as a workaround with 'get_swagger_ui_html', it is also ignored unless I add 'openapi_url' when instantiating the app. But that of course breaks the whole swagger

### Operating System

macOS

### FastAPI Version

0.112.1

### Pydantic Version

2.8.2

### Python Version

3.10.14

**Comment by YuriiMotov** [comment, marked as answer]
You can't override existing endpoint.

You should disable internal `docs` endpoint by using `docs_url=None`:

```
app = FastAPI(docs_url=None)
```

https://fastapi.tiangolo.com/how-to/custom-docs-ui-assets/#disable-the-automatic-docs

**Reply by fritz-trawa** [reply]
Excellent @YuriiMotov, thank you very much
